src/integration-test/java/com/commercetools/sync/integration/sdk2/externalsource/states/StateSyncIT.java

```diff
@@ -0,0 +1,1285 @@
+            .type(StateTypeEnum.LINE_ITEM_STATE)
+            .name(ofEnglish("state-name"))
+            .description(ofEnglish("state-desc"))
+            .roles(List.of(StateRoleEnum.RETURN))
+            .initial(false)
+            .build();
+
+    CTP_TARGET_CLIENT.states().create(stateDraft).executeBlocking();
+    errorCallBackMessages = new ArrayList<>();
+    warningCallBackMessages = new ArrayList<>();
+    errorCallBackExceptions = new ArrayList<>();
+    referenceIdToKeyCache = new CaffeineReferenceIdToKeyCacheImpl();
+  }
+
+  @Test
+  void sync_withNewState_shouldCreateState() {
+    final StateDraft stateDraft =
+        StateDraftBuilder.of()
+            .key("new-state")
+            .type(StateTypeEnum.REVIEW_STATE)
+            .roles(List.of(StateRoleEnum.REVIEW_INCLUDED_IN_STATISTICS))
+            .build();
+
+    final StateSyncOptions stateSyncOptions = StateSyncOptionsBuilder.of(CTP_TARGET_CLIENT).build();
+
+    final StateSync stateSync = new StateSync(stateSyncOptions);
+
+    // test
+    final StateSyncStatistics stateSyncStatistics =
+        stateSync.sync(List.of(stateDraft)).toCompletableFuture().join();
+
+    assertThat(stateSyncStatistics).hasValues(1, 1, 0, 0, 0);
+  }
+
+  @Test
+  void sync_withCreateStateException_shouldPrintMessage() {
+    final StateDraft stateDraft =
+        StateDraftBuilder.of()
+            .key(keyA)
+            .type(StateTypeEnum.REVIEW_STATE)
+            .roles(List.of(StateRoleEnum.REVIEW_INCLUDED_IN_STATISTICS))
+            .build();
+
+    final ProjectApiRoot client = buildClientWithBadGatewayExceptionOnPost();
+
+    final StateSyncOptions stateSyncOptions =
+        StateSyncOptionsBuilder.of(client)
+            .errorCallback(
+                (exception, oldResource, newResource, updateActions) -> {
+                  errorCallBackMessages.add(exception.getMessage());
+                  errorCallBackExceptions.add(exception.getCause());
+                })
+            .build();
+
+    final StateSync stateSync = new StateSync(stateSyncOptions);
+
+    // test
+    final StateSyncStatistics stateSyncStatistics =
+        stateSync.sync(List.of(stateDraft)).toCompletableFuture().join();
+
+    assertThat(stateSyncStatistics).hasValues(1, 0, 0, 1, 0);
+    assertThat(errorCallBackExceptions).isNotEmpty();
+    assertThat(errorCallBackMessages).isNotEmpty();
+    assertThat(errorCallBackMessages.get(0))
+        .contains(format("Failed to create draft with key: '%s'", keyA));
+  }
+
+  @Test
+  void sync_withNewStateWithoutRole_shouldRemoveRole() {
+    final StateDraft stateDraft =
+        StateDraftBuilder.of().key(keyA).type(StateTypeEnum.REVIEW_STATE).roles(List.of()).build();
+
+    final StateDraft stateDraftTarget =
+        StateDraftBuilder.of()
+            .key(keyA)
+            .type(StateTypeEnum.REVIEW_STATE)
+            .roles(List.of(StateRoleEnum.REVIEW_INCLUDED_IN_STATISTICS))
+            .build();
+    createStateInTarget(stateDraftTarget);
```

Comment:
why do you need another state in target with `keyA` ?

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/1043/commits/542ff2af53cf56aaccdd8cd76b645c848672abe3